enumerable.rb

```diff
@@ -0,0 +1,131 @@
+module Enumerable
+  def my_each
+    return to_module unless block_given?
+
+    x = 0
+    while x < length
+      yield(to_a[x])
+      x += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_module unless block_given?
+
+    x = 0
+    while x < length
+      yield(self[x], x)
+      x += 1
+    end
+    self
+  end
+
+  def my_sitemct
+    return to_module(:my_each) unless block_given?
+
+    arr = []
+    x = 0
+    while x < length
+      arr << self[x] if yield(self[x])
+      x += 1
+    end
+    arr
+  end
+
+  def my_all?(items = nil)
+    if block_given?
+      my_each { |x| return false unless yield x }
+    else
+      return my_all? { |obj| obj } unless items
+
+      if items.class == Regexp
+        my_each { |x| return false unless items.match?(x) }
+      elsif items.class == Class
+        my_each { |x| return false unless x.is_a? items }
+      else
+        my_each { |x| return false unless x == items }
+      end
+    end
+    true
+  end
+
+  def my_any?(items = nil)
+    result = false
+    if block_given?
+      my_each { |item| result = true if yield item }
+    elsif items
+      my_each { |item| result = true if itemstern?(item, items) }
+    else
+      my_each { |item| result = true if item }
+    end
+    result
+  end
+
+  def my_none?(items = nil)
+    result = true
+    if block_given?
+      my_each { |item| result = false if yield item }
+    elsif items
+      my_each { |item| result = false if itemstern?(item, items) }
+    else
+      my_each { |item| result = false if item }
+    end
+    result
+  end
+
+  def my_count(arg = nil)
+    count = 0
+    if block_given?
+      my_each { |x| count += 1 if yield(x) }
+    elsif arg
+      my_each { |x| count += 1 if x == arg }
+    else
+      count = length
+    end
+    count
+  end
+
+  def my_map(param = nil)
+    return to_module unless block_given?
+
+    new_arr = []
+    if block_given?
+      my_each { |x| new_arr << yield(x) }
+    else
+      my_each { |x| new_arr << param.call(x) }
+    end
+    new_arr
+  end
```

Comment:
* [ ] should have working `my_map`  identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
#### Errors
```bash
rspec ./spec/spec_enumerables_spec.rb:200 # enumerables my_map returns an Enumerator if no block is given
```